src/java_bytecode/java_bytecode_convert_method.cpp

```diff
@@ -232,6 +232,14 @@ void java_bytecode_convert_method_lazy(
   method_symbol.mode=ID_java;
   method_symbol.location=m.source_location;
   method_symbol.location.set_function(method_identifier);
+  if(m.is_public)
+    member_type.set(ID_access, ID_public);
```

Comment:
What I would like to avoid is having `typet` objects floating around that have `ID_access` set without anything ever making use of that. Notably, would equality comparison succeed on two `typet` with `ID_access` set to different values -- likely not. Thus such information should either be communicated via "#access" ("ID_C_access") or with objects of proper type and API. Maybe it takes introducing `java_typet`, but then this needs to propagate to, e.g., `base_type_eq`.